chapters/functions.tex

```diff
@@ -1567,13 +1567,21 @@ \section{Declaring Inverses of Functions}\label{declaring-inverses-of-functions}
 
 The function in the \lstinline!inverse! annotation must be an actual inverse.
 This requires that $f_2$ is such that, if $u_k$ is calculated as \lstinline!$u_k$ := $f_2$($\ldots$, y, $\ldots$)!, the equality \lstinline!y = $f_1$($\ldots$, $u_k$, $\ldots$)! is satisfied up to a certain precision, for all valid values of the input arguments of \lstinline!$f_2$($\ldots$, y, $\ldots$)!.
-the equality \lstinline!y = $f_1$($\ldots$, $u_k$, $\ldots$)! is satisfied up to a certain precision.
+
+\begin{nonnormative}
+There is no guarantee that a provided inverse will be used, and no rule for at which stage of symbolic processing it could be applied.
+Hence, providing an inverse function cannot be used as a means to select a particular solution to a nonlinear equation with multiple solutions (in other words, to select particular pre-image for a non-injective function).
+On the contrary, providing an inverse function can make a model ambiguous, as it won't be known if an equation is solved using a generic nonlinear equation solver (with solution being influenced by guess values), or solved using the provided inverse (possibly selecting a different solution).
```

Comment:
I don't see how this make the model "ambiguous", as this only applies if the equation already have multiple solutions.
(For "won't" see comment below).

I would either just remove it or rewrite it as:

Providing an inverse function can even make a model result more ambiguous in some cases.
This can occur if the model equations have multiple solutions, since the inverse can select a different solution than the generic nonlinear equation solver (whose solution is influenced by guess values).
However, if the original function is bijective this is not an issue.

Reply:
The point is that nonlinear equations with multiple solutions is a kind of ambiguity that it is standard to handle by providing guess values near the desired solution.  The kind of ambiguity added by inverse functions is worse, since it puts the guess value control out of play.

Reply:
Yes, but the guess value isn't always fully reliable since it depends on the solver method and its region of convergence. The point was that even if an inverse can make this worse it cannot create it from thin air.

Reply:
Sure, it's not a fully reliable mechanism, yet…

But regarding the fast that problems don't come out of thin air, I suggest that we reconsider the decision to allow inverse functions when there are multiple possible solutions, see comment above.

Reply:
I still think that this should be corrected to indicate that the underlying problem is that there are multiple inverses to the function (or in other words that the function is not injective). Specifying inverses may make this worse. If we want to have a better solution for this we add some mechanism for sending in guess-values for the unknown to the inverse; but that's for the future.

Reply:
I realize that this conversation is now being continued [below](#discussion_r604077247).  Leaving it as unresolved for now, so that comments remain visible, but please don't add more comments here.

Reply:
Now that the continued discussion has been resolved, it's time to also resolve this one.